src/app/machines/views/MachineList/MachineListHeader/MachineListHeader.test.tsx

```diff
@@ -102,65 +110,144 @@ describe("MachineListHeader", () => {
+    state.machine.selectedMachines = { items: ["abc123"] };
+    state.machine.counts["mocked-nanoid-2"] = machineStateCountFactory({
+      count: 10,
+      loaded: true,
+    });
+    state.machine.counts["mocked-nanoid-3"] = machineStateCountFactory({
+      count: 2,
+      loaded: true,
+    });
+    renderWithBrowserRouter(
+      <MachineListHeader
+        headerContent={null}
+        searchFilter=""
+        setHeaderContent={jest.fn()}
+        setSearchFilter={jest.fn()}
+      />,
+      { wrapperProps: { state } }
     );
-    wrapper
-      .find('[data-testid="section-header-subtitle"] Button')
-      .simulate("click");
-    expect(setSearchFilter).toHaveBeenCalledWith("in:(Selected)");
+    expect(screen.getByText("1 of 10 machines selected")).toBeInTheDocument();
   });
 
-  it("displays a message when all machines have been selected", () => {
-    state.machine.loaded = true;
-    // TODO: This state can be remove once the count has been updated to use the
-    // new API:
-    // https://github.com/canonical/app-tribe/issues/1102
-    state.machine.selected = ["abc123", "def456"];
-    state.machine.selectedMachines = { items: ["abc123", "def456"] };
-    const store = mockStore(state);
-    const wrapper = mount(
-      <Provider store={store}>
-        <MemoryRouter
-          initialEntries={[{ pathname: "/machines", key: "testKey" }]}
-        >
-          <CompatRouter>
-            <MachineListHeader
-              headerContent={null}
-              setHeaderContent={jest.fn()}
-              setSearchFilter={jest.fn()}
-            />
-          </CompatRouter>
-        </MemoryRouter>
-      </Provider>
+  it("displays a selected count if some groups have been selected", () => {
+    state.machine.selectedMachines = {
+      groups: ["admin"],
+      grouping: FetchGroupKey.Owner,
+    };
+    state.machine.counts["mocked-nanoid-2"] = machineStateCountFactory({
+      count: 10,
+      loaded: true,
+    });
+    state.machine.counts["mocked-nanoid-3"] = machineStateCountFactory({
+      count: 2,
+      loaded: true,
+    });
+    renderWithBrowserRouter(
+      <MachineListHeader
+        headerContent={null}
+        searchFilter=""
+        setHeaderContent={jest.fn()}
+        setSearchFilter={jest.fn()}
+      />,
+      { wrapperProps: { state } }
     );
-    expect(wrapper.find('[data-testid="section-header-subtitle"]').text()).toBe(
-      "All machines selected"
+    expect(screen.getByText("2 of 10 machines selected")).toBeInTheDocument();
+  });
+
+  it("displays a selected count if some machines and groups have been selected", () => {
+    state.machine.selectedMachines = {
+      items: ["abc123"],
+      groups: ["admin"],
+      grouping: FetchGroupKey.Owner,
+    };
+    state.machine.counts["mocked-nanoid-2"] = machineStateCountFactory({
+      count: 10,
+      loaded: true,
+    });
+    state.machine.counts["mocked-nanoid-3"] = machineStateCountFactory({
```

Comment:
Might be worth using a variable as I suggested above and/or adding a comment explaining that `mocked-nanoid-3` refers to the count of machines in a selected group.